test/driver.js

```diff
@@ -479,7 +479,22 @@ class Driver {
           enableXfa: task.enableXfa,
           styleElement: xfaStyleElement,
         });
-        loadingTask.promise.then(
+        let promise = loadingTask.promise;
+
+        if (task.save && task.annotationStorage) {
+          promise = loadingTask.promise.then(async doc => {
```

Comment:
Is it ever expected that `save` is set but an `annotationStorage` isn't provided in the manifest?
If not, we should probably make that case an error:
```suggestion
        if (task.save) {
          if (!task.annotationStorage) {
            promise = Promise.reject("Missing `annotationStorage` entry.");
          } else {
            promise = loadingTask.promise.then(async doc => {
              ...
```